File: cpp/Echo.cpp
#include <string>
#include <vector>
#include "Echo.h"
string EchoImpl::echo (string msg)
{
    return msg;
}

string EchoImpl::concat (vector<string> msgs)
{
    string ret;
    for (vector<string>::iterator i = msgs.begin(); i != msg.end(); i++) {
        ret += *i;
    }
    return ret;
}

File: java/Echo.java
import java.util.*;

interface Echo {
    String echo (String msg);
    String concat (List words);
    void touchmenot () throws TestException;
    void recursiveEcho (String msg, EchoCallback cb);
    void hello (Person p);
    void asyncEcho (String msg, EchoCallback cb);
}
